manifests/candlepin.pp

```diff
@@ -122,7 +128,11 @@
       mode   => '0750',
     } ~>
     exec { 'import CA into Candlepin truststore':
-      command => "keytool -import -v -keystore ${amqp_truststore} -storepass ${keystore_password} -alias ${alias} -file ${ca_cert} -noprompt",
+      command => "keytool -import -trustcacerts -v -keystore ${keystore} -storepass ${keystore_password} -alias ${alias} -file ${ca_cert} -noprompt",
+      unless  => "keytool -list -keystore ${keystore} -storepass ${keystore_password} -alias ${alias}",
+    } ~>
+    exec { 'import CA into Candlepin AMQP truststore':
+      command => "keytool -import -v -keystore ${amqp_truststore} -storepass ${keystore_password} -alias ${alias} -file ${ca_cert} -trustcacerts -noprompt",
       creates => $amqp_truststore,
```

Comment:
It wasn't here before, but maybe this also needs an unless instead of creates so it's also updated. I like your new approach.